### DISCUSSION TITLE
OpenSSL 3.1.x FIPS functional testing

### DISCUSSION THREAD

**Discussion by pgarba** [openssl/openssl#24644] [Q&A]
Hi,

We try to certify OpenSSL 3.1.x for FIPS 140-3 on our own and we were wondering where the functional test suite has gone.

In older versions there was a FIPS test suite to do the functional testing:
`https://github.com/majek/openssl/blob/master/fips/fips_test_suite.c`

Is there a way to tigger those kind of tests in the current OpenSSL version or does anyone have some hints how it can be done ?

Thanks,
Peter

**Comment by paulidale** [comment]
The 3.1 release stream releases are not FIPS validated. You are literally on your own here.

The **only** FIPS approved releases are 3.0.8 and 3.0.9.  Use **anything** else are you cannot be FIPS approved.

**Reply by pgarba** [reply]
I'm surprised because the announcements earlier (https://www.openssl.org/blog/blog/2023/03/07/OpenSSL3.1Release/index.html and https://www.openssl.org/blog/blog/2022/09/30/fips-140-3/index.html) said that OpenSSL 3.1 is FIPS 140-3 compliant.

**Reply by paulidale** [reply]
The 140-3 validation is a long long long way off.  Our FIPS lab was de-registered a while ago (check _Acumen_ via the NIST web site) and I've no idea what mitigation the projects intends. to do going forwards.

It seems likely that the 140-3 validation is back to early steps, but I'm out of the loop and don't know for certain.

**Reply by pgarba** [reply]
Thanks for the update. 

We already started the certification so I guess we will discover during the way what needs to be updated on our own.

But coming back to the initial question. Do you know if there is a replacement for the old `fips_test_suite.c` or is it still an item on the roadmap ?
